code/modules/paperwork/fax.dm

```diff
@@ -12,34 +12,92 @@
 	var/fax_id
 	/// The name of the fax displayed in the list. Not necessarily unique to some EMAG jokes.
 	var/fax_name
-	/// A reference to an `/obj/item/paper` inside the copier, if one is inserted. Otherwise null.
-	var/obj/item/paper/paper_contain
+	/// A weak reference to an inserted object.
+	var/datum/weakref/loaded_item_ref
+	/// True if the fax machine should be visible to other fax machines in general.
+	var/visible_to_network = TRUE
 	/// Necessary to hide syndicate faxes from the general list. Doesn't mean he's EMAGGED!
 	var/syndicate_network = FALSE
 	/// This is where the dispatch and reception history for each fax is stored.
 	var/list/fax_history = list()
+	/// World ticks the machine is electified for.
+	var/seconds_electrified = MACHINE_NOT_ELECTRIFIED
+	/// If true we will eject faxes at speed rather than sedately place them into a tray.
+	var/hurl_contents = FALSE
+	/// If true you can fax things which strictly speaking are not paper.
+	var/allow_exotic_faxes = FALSE
+	/// If true, the fax machine is jammed and needs cleaning
+	var/jammed = FALSE
+	/// List of types which should always be allowed to be faxed
+	var/static/list/allowed_types = list(
+		/obj/item/paper,
+		/obj/item/photo,
+		/obj/item/tcgcard
+	)
+	/// List of types which should be allowed to be faxed if hacked
+	var/static/list/exotic_types = list(
+		/obj/item/food/pizzaslice,
+		/obj/item/food/root_flatbread,
+		/obj/item/food/salami,
+		/obj/item/throwing_star,
+		/obj/item/stack/spacecash,
+		/obj/item/holochip,
+		/obj/item/card
+	)
 
 /obj/machinery/fax/Initialize(mapload)
 	. = ..()
 	if (!fax_id)
 		fax_id = SSnetworks.assign_random_name()
 	if (!fax_name)
 		fax_name = "Unregistered fax " + fax_id
+	wires = new /datum/wires/fax(src)
+	register_context()
 
 /obj/machinery/fax/Destroy()
-	QDEL_NULL(paper_contain)
+	QDEL_NULL(loaded_item_ref)
+	QDEL_NULL(wires)
 	return ..()
 
-/obj/machinery/fax/update_icon_state()
-	if(paper_contain)
-		icon_state = "fax_contain"
-	else
-		icon_state = "fax"
+/obj/machinery/fax/update_overlays()
+	. = ..()
+	if (panel_open)
+		. += "fax_panel"
+	var/obj/item/loaded = loaded_item_ref?.resolve()
+	if (loaded)
+		. += mutable_appearance(icon, find_overlay_state(loaded, "contain"))
+
+/obj/machinery/fax/examine()
+	. = ..()
+	if(jammed)
+		. += span_notice("Its output port is jammed and needs cleaning.")
+
+
+/obj/machinery/fax/on_set_is_operational(old_value)
+	if (old_value == FALSE)
```

Comment:
```suggestion
	if (!old_value)
```

Reply:
I decided to make it explicit `== FALSE` here because I feel like it just is easier to read that way, but I'll change it if it's required.

Reply:
That's fair I suppose, especially since ``old_value`` is a very uninformative var name.